pkg/gofr/container/datasources.go

```diff
@@ -266,3 +267,24 @@ type KVStoreProvider interface {
 
 	provider
 }
+
+type Solr interface {
+	Search(ctx context.Context, collection string, params map[string]interface{}) (interface{}, error)
```

Comment:
Please use any, here and everywhere


```suggestion
	Search(ctx context.Context, collection string, params map[string]any) (any, error)
```